### PR TITLE
Simplifying bicep modules

### LINKED ISSUE #831: [azdev-ify] Module discovery and usage for /core
It's not easy to figure out what are the modules I should be using - especially for hosts.

e.g., In /host
- we have 11 biceps for appservice
- 4 for ACA (container-app.bicep vs. container-apps.bicep - what's the diff?) 
- 3 for functions - more straightforward - a base and 2 lang-specific.
- 1 for staticwebapp

Consider:
**Scenario 1 - I just want to use what's in /core**
All I need is to know is what parameters to pass to the modules, I may not need to know what's in the modules but I need to know what each module is for and what is the contract.

**Scenario 2 - I want to modify what's in /core and create my own private collection**
For this scenario, developers do need to understand what's in the modules figure out how to customize/add own logics/conventions.

### PR DESCRIPTION
**Pull request by jongio**
Fixes #831

1. Reorganized modules in /core to be in subfolders
2. Removed lang specific appservice and functions files
3. Make KV optional for app service
4. Refactor and rearrange some params
5. Added enable_oryx_build setting

### CODE REVIEW

**Review by wbreza — COMMENTED** [review]
This is looking great - main concern is the SKU options that are now required when specifying the ASP for each template.  See my other comments for more details.

File: templates/todo/projects/csharp-sql/.repo/bicep/infra/resources.bicep [hunk 1]
```diff
@@ -53,11 +65,14 @@ module apiSqlServerConfig '../../../../../../common/infra/bicep/core/host/appser
...
 // Create an App Service Plan to group applications under the same payment plan and SKU
-module appServicePlan '../../../../../../common/infra/bicep/core/host/appserviceplan-sites.bicep' = {
+module appServicePlan '../../../../../../common/infra/bicep/core/host/appservice/appserviceplan.bicep' = {
   name: 'appserviceplan'
   params: {
     environmentName: environmentName
     location: location
+    sku: {
+      name: 'B1'
+    }
```
[the remark this answers is not in the document]
Comment (jongio) [reply templates/todo/projects/csharp-sql/.repo/bicep/infra/resources.bicep:75]: I agree that's there's improvements to be made here, but will take as a different change.  I personally think we need to make SKUs easier to figure out versus hiding them.  Need to think on it.

The challenge is that an app service plan for sites is a lot diff than for functions. 

1. Sometimes you have both function and appservice in the same plan. So we need to flesh out what makes sense from ux there.
2. Sometimes you want to share the app service plan.  Which one creates it? How do you handle the dependency between them?  Need to think it through more.

File: templates/todo/projects/csharp-cosmos-sql/.repo/bicep/infra/resources.bicep [hunk 2]
```diff
@@ -49,11 +60,14 @@ module apiCosmosConfig '../../../../../../common/infra/bicep/core/host/appservic
...
 // Create an App Service Plan to group applications under the same payment plan and SKU
-module appServicePlan '../../../../../../common/infra/bicep/core/host/appserviceplan-sites.bicep' = {
+module appServicePlan '../../../../../../common/infra/bicep/core/host/appservice/appserviceplan.bicep' = {
   name: 'appserviceplan'
   params: {
     environmentName: environmentName
     location: location
+    sku: {
+      name: 'B1'
+    }
```
[the remark this answers is not in the document]
Comment (jongio) [reply templates/todo/projects/csharp-cosmos-sql/.repo/bicep/infra/resources.bicep:70]: Marking this as same as other comment.

**Review by wbreza — APPROVED** [review]
Spoke with Jon offline and agreed to hold off on SKU changes for now.  Looks great!

**Pull request merged**